app/src/main/java/org/sopt/sample/SignUpActivity.kt

```diff
@@ -0,0 +1,35 @@
+package org.sopt.sample
+
+import android.content.Intent
+import android.os.Bundle
+import android.os.PersistableBundle
+import androidx.appcompat.app.AppCompatActivity
+import org.sopt.sample.databinding.ActivityLoginBinding
+import com.google.android.material.snackbar.Snackbar
+import org.sopt.sample.databinding.ActivitySignupBinding
+
+class SignUpActivity: AppCompatActivity(){
+    private lateinit var binding: ActivitySignupBinding
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivitySignupBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        binding.btSignupend.setOnClickListener() {
+            if (binding.etId.text.length >= 6 && binding.etId.text.length <= 10) {
+                if (binding.etPw.text.length >= 8 && binding.etPw.text.length <= 12) {
+                    Snackbar.make(binding.root, "회원가입이 완료되었습니다.", Snackbar.LENGTH_SHORT).show()
+                    val intent = Intent(this, LoginActivity::class.java)
+                    intent.putExtra("id", binding.etId.text.toString())
+                    intent.putExtra("pw", binding.etPw.text.toString())
+                    intent.putExtra("mbti", binding.etMbti.text.toString())
+                    startActivity(intent)
```

Comment:
이 로직은 버튼을 클릭 했을 때 내용인데 onCreate()에 모두 넣기 보다는! 따로 함수로 빼둬서 정리해보는 것은 어떨까요!
ex) private fun clickEvent() { 내용 }

Reply:
조금이나마 binding 중복을 줄이기 위해서 with(binding)으로 바인딩을 묶을 수 있습니다! 

```suggestion
with(binding){
        btSignupend.setOnClickListener() {
            if (etId.text.length >= 6 && etId.text.length <= 10) {
                if (.etPw.text.length >= 8 && etPw.text.length <= 12) {
                    Snackbar.make(binding.root, "회원가입이 완료되었습니다.", Snackbar.LENGTH_SHORT).show()
                    val intent = Intent(this, LoginActivity::class.java)
                    intent.putExtra("id", etId.text.toString())
                    intent.putExtra("pw", etPw.text.toString())
                    intent.putExtra("mbti", etMbti.text.toString())
                    startActivity(intent)
                   ...
    }
                    
```